db.py

```diff
@@ -84,6 +82,15 @@ class Researcher(db.Model):
     user = db.relationship('User', backref=db.backref('user_researcher', uselist=False))
 
 
+class ResearcherSupervisor(db.Model):
+    __tablename__ = 'researcher_supervisor'
+    researcher_id = db.Column(db.Integer, db.ForeignKey('researcher.id'), primary_key=True)
+    supervisor_id = db.Column(db.Integer, db.ForeignKey('user.id'), primary_key=True)
+
+    researcher = db.relationship('Researcher', backref=db.backref('researcher_supervisor', lazy=True))
```

Comment:
Why don't you simply call the backref `supervisors` so that it is more intuitive to fetch `Researcher.supervisors` to get the researcher supervisors ?